htdocs/class/XoopsTokenHandler.php

```
@@ -0,0 +1,227 @@
+     * @param int    $uid            User ID
+     * @param string $scope          Token scope (e.g. 'lostpass', 'activation')
+     * @param int    $ttl            Time-to-live in seconds (minimum 60)
+     * @param bool   $revokePrevious Revoke unused tokens for same scope first
+     *
+     * @return string|false Raw token string, or false on DB failure
+     */
+    public function create(
+        int    $uid,
+        string $scope,
+        int    $ttl = 3600,
+        bool   $revokePrevious = true
```

PHPMD (2.15.0)

[error] The method create has a boolean flag argument $revokePrevious, which is a certain sign of a Single Responsibility Principle violation. (undefined)

(BooleanArgumentFlag)